src/Http/Handlers/AuthenticateHandler.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace OpenIDConnectServer\Http\Handlers;
+
+use OAuth2\Request;
+use OAuth2\Response;
+use OpenIDConnectServer\Http\RequestHandler;
+use OpenIDConnectServer\Http\Router;
+use OpenIDConnectServer\Storage\ConsentStorage;
+
+class AuthenticateHandler extends RequestHandler {
+	private ConsentStorage $consent_storage;
+	private array $clients;
+
+	public function __construct( ConsentStorage $consent_storage, array $clients ) {
+		$this->clients         = $clients;
+		$this->consent_storage = $consent_storage;
+	}
+
+	public function handle( Request $request, Response $response ): Response {
+		$client_name = $this->get_client_name( $request );
+		if ( ! $client_name ) {
+			$response->setStatusCode( 404 );
+
+			return $response;
+		}
+
+		if ( ! is_user_logged_in() ) {
+			auth_redirect();
+		}
+
+		if ( $this->consent_storage->needs_consent( get_current_user_id() ) ) {
+			define( 'OIDC_DISPLAY_AUTHORIZE', true );
+			include __DIR__ . '/../../Template/Authorize.php';
+		} else {
+			// rebuild request with all parameters and send to authorize endpoint.
+			wp_safe_redirect(
+				add_query_arg(
+					array_merge(
+						array( '_wpnonce' => wp_create_nonce( 'wp_rest' ) ),
+						$request->getAllQueryParameters()
+					),
+					Router::make_rest_url( 'authorize' )
+				)
+			);
+		}
+
+		// TODO: return response instead of exiting.
+		exit;
+	}
+
+	/**
+	 * TODO: Remove this function in favour of ClientCredentialsStorage?
+	 */
+	private function get_client_name( Request $request ): string {
+		$client_id = $request->query( 'client_id' );
+
+		if ( ! array_key_exists( $client_id, $this->clients ) ) {
```

Comment:
I tend to prefer `array_key_exists()` as I find it's more explicit than `isset()`, since `isset()` also checks for `null` (as you mentioned), which you don't always want. So sticking to `array_key_exists()` makes it for one fewer decisions to make :)

It also triggers me a little bit that you don't get a warning when trying to access a non-existent key, which you would in any other situation. I.e. `$foo['non_existent']` triggers a warning, while `isset( $foo['non_existent'] )` doesn't. So some magic is happening at the language level, which, I just noticed, is stated in the [docs](https://www.php.net/manual/en/function.isset.php):

> Note: Because this is a language construct and not a function, it cannot be called using [variable functions](https://www.php.net/manual/en/functions.variable-functions.php), or [named arguments](https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments).

So `isset()` comes with some extra limitations of when you can use it, which ties to the fewer decisions thing I mentioned.

That said, I'm fine with using `isset()` here. I pushed a commit that removes all calls to `array_key_exists()` in favour of `isset()`, and I'll stick to `isset()` here from now on.